src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateHistory = [];
+  let currentState = { ...state }; // Start with a clone of the initial state
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
+
+        action.keysToRemove.forEach((key) => {
+          delete currentState[key];
+        });
+        break;
+      case 'clear':
+        currentState = {};
+        break;
+      default:
+        throw new Error(`Unknown action type: ${action.type}`);
+    }
+    stateHistory.push(currentState);
```

Comment:
You are pushing the reference of `currentState` into the `stateHistory` array. This will result in all elements of `stateHistory` pointing to the same object. Instead, you should push a copy of `currentState` to ensure each element in the history array is a unique state.